vault/resource_userpass_user.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func userpassUserResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		consts.FieldMount: {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Description:  "Auth mount at which user will be configured.",
+			Default:      consts.MountTypeUserpass,
+			ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+		},
+		consts.FieldUsername: {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "Username part of userpass.",
+			ValidateFunc: provider.ValidateStringSlug,
+		},
+		consts.FieldPassword: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Sensitive:   true,
+			ForceNew:    true,
```

Comment:
I don't think this needs to be a `ForceNew`, but I could be wrong. If I understand correctly, a created user with a particular username should be able to update their password, is that not the case? If the password can be updated for a user, then we can remove this